src/arrayMethodJoin.js

```diff
@@ -1,11 +1,41 @@
 'use strict';
 
-/**
+/*
  * Implement method join
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+
+    let newStr = `${this[0]}`;
+
+    if (this[0] === undefined || this[0] === null) {
+      newStr = '';
+    }
```

Comment:
It seems, that you complicate the task when you assign a value to a string as this[0]. If it is empty - you have to do an extra check, in your case - you have 2 extra checks.

Reply:
I change the logic, it should be fine now.